include/crocoddyl/multibody/actions/contact-fwddyn.hxx

```diff
@@ -82,15 +83,16 @@ void DifferentialActionModelContactFwdDynamicsTpl<Scalar>::calc(
   contacts_->calc(d->multibody.contacts, x);
 
 #ifndef NDEBUG
-  Eigen::FullPivLU<MatrixXs> Jc_lu(d->multibody.contacts->Jc);
+  Eigen::FullPivLU<MatrixXs> Jc_lu(d->multibody.contacts->Jc.topRows(nc));
```

Comment:
Isn't the size of Jc already nc?

Reply:
I have introduced, in a previous PR, the concept on active and inactive contacts. `nc_total` defines the dimension of total number of contacts, instead `nc` defines the dimension of active contacts. The same has been done for impulses and costs.

The idea is to allow user to allocate the memory only once. For instance, you can create a contact dynamics with all contact consider in your MPC, and then online to define which of them are active.

Saying that, the answer of your question is no. `Jc` is defined with `nc_total` but we fill the first `nc` rows (active contacts).

Let me know if you have more questions regarding this topic. Btw, I have tested in my MPC formulation. Later I will make it public.